src/insert.rs

```diff
@@ -60,62 +68,20 @@ impl<T> Insert<T> {
     where
```

Comment:
So, we're moving almost all checks into `fn write()` here. It's (usually) not a problem, but a notable change that should be noted in the changelog.

In the following breaking release, we can return `-> Self` instead of `Result`. Thus, we should at least put `TODO: remove Result` here so as not to forget it.

Reply:
Maybe we should keep the Result here anyway, without removing it. It will allow more flexibility if some additional logic will have to unwrap a result in the `new` in the future.

Reply:
Added a note to the changelog